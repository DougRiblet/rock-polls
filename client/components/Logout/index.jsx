import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { logOutUser } from '../../actions/actions';
import logoutPresent from './logout-presentational';

const mapStateToProps = state => ({
  authenticated: state.authenticated,
  user_id: state.user_id,
  username: state.username,
});

const mapDispatchToProps = dispatch => bindActionCreators({ logOutUser }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(logoutPresent);
