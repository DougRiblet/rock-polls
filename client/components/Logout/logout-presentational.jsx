// @flow

import React from 'react';

type Props = {
  logOutUser: () => mixed,
  authenticated: boolean,
  username: string,
};

export default class Logout extends React.Component<Props> {
  constructor(props: Props) {
    super(props);
    this.handleLogout = this.handleLogout.bind(this);
  }

  handleLogOut: Function;

  // eslint-disable-next-line no-undef
  handleLogout() {
    this.props.logOutUser();
  }

  render() {
    return (
      <div id='logout'>
        <div>
          <button onClick={this.handleLogout}>
            Log Out
          </button>
        </div>
        <div className='show-status'>
          <p>Authenticated: {String(this.props.authenticated)}</p>
          <p>Username: {this.props.username}</p>
        </div>
      </div>
    );
  }
}
