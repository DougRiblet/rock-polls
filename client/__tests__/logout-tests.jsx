import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Logout from '../components/Logout/logout-presentational';

describe('Logout', () => {
  test('Snapshot', () => {
    const Logout_tree = renderer.create(<Logout
      logOutUser={() => logOutUser()}
      authenticated={false}
      user_id={''}
      username={''}
    />).toJSON();
    expect(Logout_tree).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Logout
      logOutUser={() => logOutUser()}
      authenticated={false}
      user_id={''}
      username={''}
    />, div);
  });

});
