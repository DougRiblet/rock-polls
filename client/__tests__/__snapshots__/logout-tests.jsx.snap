// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Snapshot 1`] = `
<div
  id="logout"
>
  <div>
    <button
      onClick={[Function]}
    >
      Log Out
    </button>
  </div>
  <div
    className="show-status"
  >
    <p>
      Authenticated: 
      false
    </p>
    <p>
      Username: 
      
    </p>
    <p>
      User ID: 
      
    </p>
  </div>
</div>
`;
